Content.Server/Explosion/EntitySystems/ExplosionSystem.TileFill.cs

```diff
@@ -251,6 +252,33 @@ public sealed partial class ExplosionSystem
         return (totalTiles, iterationIntensity, spaceData, gridData, spaceMatrix);
     }
 
+    private bool FindInitialTile(MapCoordinates epicenter, out Vector2i initialTile, out EntityUid? epicentreGrid)
+    {
+        initialTile = new Vector2i();
+        epicentreGrid = null;
+
+        if (!_mapManager.TryFindGridAt(epicenter, out var gridUid, out var candidateGrid))
+        {
+            return false;
+        }
+
+        var tilePos = _map.WorldToTile(gridUid, candidateGrid, epicenter.Position);
+        if (!_map.TryGetTileRef(gridUid, candidateGrid, tilePos, out var tileRef))
+        {
+            return false;
+        }
+
+        if (tileRef.Tile.IsEmpty)
+        {
+            return false;
+        }
+
+        epicentreGrid = gridUid;
+        initialTile = tileRef.GridIndices;
+
+        return true;
+    }
+
```

Comment:
This looks like it should be an engine method on SharedMapSystem if anything?

Also doesn't TryFindGridAt fail if the tile is empty? is the return at 273 even possible?